test/bases/renku_data_services/data_api/test_groups.py

```diff
@@ -0,0 +1,197 @@
+import json
+from datetime import datetime
+from test.bases.renku_data_services.keycloak_sync.test_sync import get_kc_users
+from typing import Dict, List
+
+import pytest
+import pytest_asyncio
+from sanic import Sanic
+from sanic_testing.testing import SanicASGITestClient
+
+from renku_data_services.app_config import Config
+from renku_data_services.data_api.app import register_all_handlers
+from renku_data_services.users.dummy_kc_api import DummyKeycloakAPI
+from renku_data_services.users.models import UserInfo
+
+
+@pytest.fixture
+def admin_user() -> UserInfo:
+    return UserInfo("admin", "Admin", "Doe", "admin.doe@gmail.com")
+
+
+@pytest.fixture
+def regular_user() -> UserInfo:
+    return UserInfo("user", "User", "Doe", "user.doe@gmail.com")
+
+
+@pytest.fixture
+def users(regular_user, admin_user) -> List[UserInfo]:
+    return [
+        regular_user,
+        admin_user,
+        UserInfo("member-1", "Member-1", "Doe", "member-1.doe@gmail.com"),
+        UserInfo("member-2", "Member-2", "Doe", "member-2.doe@gmail.com"),
+    ]
+
+
+@pytest_asyncio.fixture
+async def sanic_client(app_config: Config, users: List[UserInfo]) -> SanicASGITestClient:
+    app_config.kc_api = DummyKeycloakAPI(users=get_kc_users(users))
+    app = Sanic(app_config.app_name)
+    app = register_all_handlers(app, app_config)
+    await app_config.kc_user_repo.initialize(app_config.kc_api)
+    await app_config.group_repo.generate_user_namespaces()
+    return SanicASGITestClient(app)
```

Comment:
we should probably move this to common fixtures, it's duplicated in 4 places

Reply:
I opened a followup issue for this #161